core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -2451,12 +2451,14 @@ private void checkNullableTimestamp(CalciteAssert.Config config) {
     CalciteAssert.hr()
         .query(
             "select upper((case when \"empid\">\"deptno\"*10 then 'y' else null end)) T from \"hr\".\"emps\"")
-        .planContains("static final String "
-            + "$L4J$C$org_apache_calcite_runtime_SqlFunctions_upper_y_ = "
-            + "org.apache.calcite.runtime.SqlFunctions.upper(\"y\");")
-        .planContains("return current.empid <= current.deptno * 10 "
-            + "? (String) null "
-            + ": $L4J$C$org_apache_calcite_runtime_SqlFunctions_upper_y_;")
+        .planContains("      String case_when_value;\n"
```

Comment:
My understanding is that this and the following tests having the prefix `testReuseExpressionWhenNullChecking` are meant to verify that certain optimizations take place when we are generating code for nulls. 

1. Are these optimizations possible with the new code generator? 
2. Are these optimizations applied already? 
3. If not can we apply them in the future?

Reply:
The new codegen framework can naturally reuses expressions, due to:
It removes the mutable state (i.e., `Map<? extends RexNode, Boolean> exprNullableMap`), and generates code strictly corresonding to RexNode in one-to-one manner. 

Therefore, the redundant code vaused by tranversing with mutable state will not be produced again in the new codegen. On the other hand, we also keep a map in `RexToLixTranslator` for same RexNode's code reuse.

Reply:
So the answer to my question is 2 since reuse is achieved by design, is that right?

Reply:
Yes, I think so. Please correct me if I missed something :).